TypeScript Compilation Errors present in Tests do not cause the build to fail

A TypeScript compilation error was [recently introduced into the codebase](https://github.com/improbable-eng/ts-protoc-gen/pull/61/files#diff-359d2c86596ce1f00d2d7b061aacfb5aR287) but the build did not flag this.

Running the file manually through `tsc` threw the expected warning:

```
jonny-mbp:service jonny$ tsc --noEmit ./grpcweb.ts 
grpcweb.ts(287,38): error TS2339: Property 'name' does not exist on type '{ (requestMessage: UnaryRequest, metadata: BrowserHeaders, callback: (error: ServiceError, respon...'.
```

--- a/package.json
+++ b/package.json
@@ -37,6 +37,7 @@
     "@types/lodash": "^4.14.106",
     "@types/mocha": "^2.2.46",
     "@types/node": "^7.0.52",
+    "@types/text-encoding": "0.0.32",
     "babel": "^6.5.2",
     "chai": "^3.5.0",
     "grpc-tools": "^1.6.6",

--- a/test/helpers/fakeGrpcTransport.ts
+++ b/test/helpers/fakeGrpcTransport.ts
@@ -4,14 +4,6 @@ import { Transport, TransportConstructor, TransportOptions } from "grpc-web-clie
 import * as _ from "lodash";
 import { TextEncoder } from "text-encoding";
 
-function frameRequest(request: Message): ArrayBufferView {
-  const bytes = request.serializeBinary();
-  const frame = new ArrayBuffer(bytes.byteLength + 5);
-  new DataView(frame, 1, 4).setUint32(0, bytes.length, false /* big endian */);
-  new Uint8Array(frame, 5).set(bytes);
-  return new Uint8Array(frame);
-}
-
 function frameResponse(request: Message): Uint8Array {
   const bytes = request.serializeBinary();
   const frame = new ArrayBuffer(bytes.byteLength + 5);
@@ -209,7 +201,7 @@ export class StubTransportBuilder {
   }
 }
 
-interface TriggerableTransport {
+export interface TriggerableTransport {
   (options: TransportOptions): Transport;
   sendHeaders(): boolean;
   sendMessages(): boolean;

--- a/test/mocha-run-suite.sh
+++ b/test/mocha-run-suite.sh
@@ -15,7 +15,7 @@ fi
 mocha \
     --reporter mocha-spec-json-output-reporter \
     --reporter-options fileName=./test/mocha-report.json \
-    --require ts-node/register \
+    --require ts-node/register/type-check \
     --require source-map-support/register \
     "${TEST_SUITE}"
 

